src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/ConvertToRecordHelpers.cs

```diff
@@ -340,41 +340,14 @@ public static (ImmutableArray<ArgumentSyntax> initializerArguments, ImmutableArr
         /// i.e. Doesn't reference any other created variables but the parameters
         /// </summary>
         private static bool IsSafeAssignment(IOperation operation)
-            => operation.WalkDownConversion() switch
+        {
+            if (operation is ILocalReferenceOperation)
```

Comment:
We don't still need to check for IMemberReferenceOperation?

Reply:
Ah right, I forgot about the case where we access static members/methods. Any other member reference or invocation should have the instance and the member as ChildOperations so if a member reference of a local we should still hit it here.

Reply:
My main concern was whether a this reference counted as a local reference or as something else.